pkg/serverless/daemon/daemon_test.go

```diff
@@ -19,18 +20,24 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/trace/testutil"
 )
 
+func TestMain(m *testing.M) {
+	origShutdownDelay := ShutdownDelay
+	ShutdownDelay = 0
+	defer func() { ShutdownDelay = origShutdownDelay }()
+	os.Exit(m.Run())
+}
+
 func TestWaitForDaemonBlocking(t *testing.T) {
 	assert := assert.New(t)
 	port := testutil.FreeTCPPort(t)
 	d := StartDaemon(fmt.Sprint("127.0.0.1:", port))
-	time.Sleep(100 * time.Millisecond)
```

Comment:
I removed the sleeps after starting the daemon and found that the tests still pass consistently.